### PR TITLE
 media queries widget test

### PR DESCRIPTION
**Pull request by mshchygol**

### CODE REVIEW

File: tests/testmediaqueries.js [hunk 1]
```diff
@@ -24,7 +31,7 @@ function testMediaQueries() {
   }
 
   if (table) {
-    rows = element.getElementsByTagName('TR');
+    rows = element.getElementsByTagName('tr');
```
Comment (vpodk) [inline tests/testmediaqueries.js]: Please rollback to upper case.

File: tests/testmediaqueries.js [hunk 2]
```diff
@@ -1,9 +1,16 @@
 /**
  * @fileoverview Defines tests for testMediaQueries.
+ * Success criterias:
+ * - element with id 'media-widget' should exist;
```
Comment (vpodk) [inline tests/testmediaqueries.js]: Element

File: tests/testmediaqueries.js [hunk 3]
```diff
@@ -1,9 +1,16 @@
...
+ * - if widget is not empty it should contain element with id
```
Comment (vpodk) [inline tests/testmediaqueries.js]: If

File: tests/testmediaqueries.js [hunk 4]
```diff
@@ -1,9 +1,16 @@
...
+ *  'media-data-table';
+ * - rows in 'media-data-table' should contain text string '@media';
```
Comment (vpodk) [inline tests/testmediaqueries.js]: Rows

File: tests/testmediaqueries.js [hunk 5]
```diff
@@ -1,9 +1,16 @@
...
+ * - if widget is empty it should contain class '.rule.media-none'.
```
Comment (vpodk) [inline tests/testmediaqueries.js]: If

Comment (mshchygol) [inline tests/testmediaqueries.js]: Fixed

File: tests/testmediaqueries.js (hunk 4 above)
Comment (mshchygol) [inline tests/testmediaqueries.js]: Fixed

File: tests/testmediaqueries.js (hunk 3 above)
Comment (mshchygol) [inline tests/testmediaqueries.js]: Fixed

File: tests/testmediaqueries.js (hunk 2 above)
Comment (mshchygol) [inline tests/testmediaqueries.js]: Fixed

File: tests/testmediaqueries.js (hunk 1 above)
Comment (mshchygol) [inline tests/testmediaqueries.js]: Fixed

**Comment by mshchygol** [comment]
+1

File: tests/testmediaqueries.js [hunk 6]
```diff
@@ -1,9 +1,16 @@
 /**
  * @fileoverview Defines tests for testMediaQueries.
+ * Success criterias:
+ * - Element with id 'media-widget' should exist;
+ * - If widget is not empty it should contain element with id
+ *  'media-data-table';
+ * - Rows in 'media-data-table' should contain text string '@media';
+ * - If widget is empty it should contain class '.rule.media-none'.
```
Comment (zalesky) [inline tests/testmediaqueries.js]: classes

File: tests/testmediaqueries.js [hunk 7]
```diff
@@ -1,9 +1,16 @@
 /**
  * @fileoverview Defines tests for testMediaQueries.
+ * Success criterias:
+ * - Element with id 'media-widget' should exist;
+ * - If widget is not empty it should contain element with id
+ *  'media-data-table';
```
Comment (zalesky) [inline tests/testmediaqueries.js]: add space

File: tests/testmediaqueries.js (hunk 6 above)
Comment (RomanFedytskyi) [inline tests/testmediaqueries.js]: fixed

File: tests/testmediaqueries.js (hunk 7 above)
Comment (RomanFedytskyi) [inline tests/testmediaqueries.js]: fixed

**Comment by zalesky** [comment]
+1

File: tests/testmediaqueries.js [hunk 8]
```diff
@@ -1,41 +1,43 @@
...
 
-
 /** @const {!RegExp} */ var MEDIA_PATTERN = /@media([^{]*)/;
 /** @const {string} */ var CSS_RULE = '.rule.media-none';
 /** @const {string} */ var TABLE_ID = 'media-data-table';
 
 
 /**
- * @return {boolean} Returns "true" if test failed.
+ * @return {boolean} Returns "false" if test passed.
```
Comment (vpodk) [inline tests/testmediaqueries.js]: Please see:

``` javascript
var failed = page.evaluate(function(method) {
  return window[method]();
}, 'testmediaqueries');
```

https://github.com/PagePortrait/Report/blob/master/build/jstest.sh#L101

**Comment by unknown4unnamed** [comment]
LGTM

**Comment by mshchygol** [comment]
LGTM

File: tests/testmediaqueries.js [hunk 9]
```diff
@@ -13,29 +20,25 @@
...
-  if (table) {
+  if (element && table) {
     rows = element.getElementsByTagName('TR');
     length = rows.length;
 
     for (; length;) {
-      if (!MEDIA_PATTERN.test(rows[--length].textContent.trim())) {
-        return true;
+      if (MEDIA_PATTERN.test(rows[--length].textContent.trim()) || rule) {
+        result = false;
```
Comment (vpodk) [inline tests/testmediaqueries.js]: Missed `break` statement.
Suggestion:

``` javascript
var result = false;

for (; loop;) {
  if (expression) {
    result = true;
    break;
  }
}

return result;
```

Comment (RomanFedytskyi) [inline tests/testmediaqueries.js]: fixed

File: tests/testmediaqueries.js [hunk 10]
```diff
@@ -38,6 +38,8 @@ function testMediaQueries() {
         break;
       }
     }
+  } else {
+    result = true;
```
Comment (AlbaMontis) [inline tests/testmediaqueries.js]: What if no media queries  found on page?

Comment (RomanFedytskyi) [inline tests/testmediaqueries.js]: fixed

File: tests/testmediaqueries.js [hunk 11]
```diff
@@ -13,29 +20,28 @@
...
 
-  if (table) {
+  if (element && table) {
     rows = element.getElementsByTagName('TR');
     length = rows.length;
 
     for (; length;) {
-      if (!MEDIA_PATTERN.test(rows[--length].textContent.trim())) {
-        return true;
+      if (!MEDIA_PATTERN.test(rows[--length].textContent.trim()) && !rule) {
```
Comment (vpodk) [inline tests/testmediaqueries.js]: Why you validate `rule` in loop?

**Comment by Alexsv7** [comment]
LGTM

**Comment by mshchygol** [comment]
LGTM

**Comment by zalesky** [comment]
So if variable `table` not exist but !(element && element.querySelector(CSS_RULE)) == true 
your test will have passed.

**Comment by vasylsemiliak** [comment]
LGTM

**Comment by Alexsv7** [comment]
LGTM

**Comment by AlbaMontis** [comment]
LGTM

File: tests/testmediaqueries.js [hunk 12]
```diff
@@ -13,29 +21,27 @@
...
-
     for (; length;) {
-      if (!MEDIA_PATTERN.test(rows[--length].textContent.trim())) {
-        return true;
+      if (MEDIA_PATTERN.test(rows[--length].textContent.trim())) {
+        result = false;
+      } else {
+        result = true;
+        break;
       }
```
Comment (ppokyd) [inline tests/testmediaqueries.js:48]:
``` javascript
if (!MEDIA_PATTERN.test(rows[--length].textContent.trim())) {
  result = true;
  break;
}
```

**Comment by vpodk** [comment]
LGTM

**Pull request merged**